samples/kvssink_gstreamer_sample.cpp

```diff
@@ -155,6 +161,10 @@ static bool resolution_supported(GstCaps *src_caps, GstCaps *query_caps_raw, Gst
 
 /* callback when eos (End of Stream) is posted on bus */
 static void eos_cb(GstElement *sink, GstMessage *message, CustomData *data) {
+    if (data->fragment_metadata_timer_id != 0) {
+        g_source_remove(data->fragment_metadata_timer_id);
```

Comment:
By explicitly setting data->fragment_metadata_timer_id to 0, you make it clear that the timer is no longer active, which can help prevent potential issues related to accessing stale timer IDs or mistakenly assuming a timer is active when it's not.